### PR TITLE
support reserved resources by guarantee field in QueueSpec

### PR DESCRIPTION
**Pull request by qiankunli**

### CODE REVIEW

**Comment by qiankunli** [comment]
需求及设计文档：https://note.youdao.com/s/CqjurNBy

**Comment by william-wang** [comment]
> 需求及设计文档：https://note.youdao.com/s/CqjurNBy 应在 webhook 中补充下 校验条件
> 
> 1. 一个queue 的guarantee 不能超过 capacity
> 2. 所有queue guarantee 和不能超过总资源
@qiankunli  It's a very useful functionality. Could  you submit a design doc here so that other contributors have chance to give comments for this feature?
https://github.com/volcano-sh/volcano/tree/master/docs/design

File: docs/design/queue-guarantee-resource-reservation-design.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Volcano Resource Reservation For Queue
+
+@[qiankunli](https://github.com/qiankunli); Oct 11rd, 2021
+
+## Motivation
+As [issue 1101](https://github.com/volcano-sh/volcano/issues/1101) mentioned, Volcano should support resource reservation
```
Comment (william-wang) [inline docs/design/queue-guarantee-resource-reservation-design.md:6]: Thanks for your design doc. Please add the following parts and restruct the doc.
1. Scope 
2. Out of Scope ( clarify the parts which you won't implement in this project. f.g. enhance vcctl/vqueue to configure gurantee)
3. Error handing
4. Features Interaction
The gurantee has relation with capability and proporiton behavior or even confliction with them.  you can write down the degin behavior in these use cases or scenarios.
let's say:
## use case1:   
The proportion is enabled and then create a new queue with a big gurantee value that can not be satisfied. what's the design behavor against it. Reject the creation or preempt other job or just let the job in this new queue waiting for releasing resource.

you can refer to this doc:
https://github.com/volcano-sh/volcano/blob/master/docs/design/job-api.md

**Comment by Thor-wl** [comment]
Any process about this PR? @qiankunli

**Pull request closed without merging**